csharp/ql/lib/semmle/code/csharp/File.qll

```diff
@@ -54,14 +54,14 @@ class File extends Container, Impl::File {
 
   /** Holds if this file is a QL test stub file. */
   pragma[noinline]
-  private predicate isStub() {
+  predicate isStub() {
     this.extractedQlTest() and
     this.getAbsolutePath().matches("%resources/stubs/%")
   }
 
   /** Holds if this file contains source code. */
   final predicate fromSource() {
-    this.getExtension() = "cs" and
+    this.getExtension() = ["cs", "cshtml"] and
```

Comment:
I wonder if we can end up with `cshtml` files in the DB outside of files referenced in `#line` pragmas.

Reply:
I don't think we can, but it is my understanding that since we map the locations in the generated `.cs` files into the `.cshtml` files, they will actually have an artificial `.cshtml` extension.